Make bind address and port configurable

osyoyu opened the pull request:
rspec-daemon users would want to use ports other than 3002.

Sidenote: rspec-daemon binds to 0.0.0.0 (all addresses) by default, but this might be considered unnecessarily open and insecure. I believe 127.0.0.1 is a better option, as typical rspec-daemon usage probably won't involve external access.

I haven't changed any defaults here.

asonas commented:
Changes looks good.
I have merged #4 to allow for changing the port via environment variables.
I want to ensure this change is included effectively. Therefore, please resolve any conflicts by setting the priority as follows: arguments > environment variables > default values.

osyoyu commented:
Rebased on main, and moved ENV.fetch calls to the Cli class. Arguments take precedence over env vars.

```
% RSPEC_DAEMON_BIND_ADDRESS=127.0.0.1 RSPEC_DAEMON_PORT=3003 bundle exec rspec-daemon --port 3004
Listening on tcp://127.0.0.1:3004
```

I've also added RSPEC_DAEMON_BIND_ADDRESS for consistency.
